pkg/monitoring/metrics.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright The Kubernetes NMState Authors.
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	ApplyTopologyTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "nmstate_apply_topology_total",
```

Comment:
If I understand currently, This metric counts the number of times the topology was applied.
If that is correct I would name it `nmstate_topology_applied_total`.

Reply:
Please add a help test. Its visible in the Grafana and OCP UI when searching for the metric. Its also useful for generation automated metrics documentation, like we have in kubevirt.

Reply:
We also have a metrics name linter that I would suggest you use in order to validate the metrics names meet the best practices. We added it to the PR tests, so issues will be detected early.

Reply:
@sradco do you have a pointer to that linter to integrate it ?

Reply:
The linter we use for kubevirt is here: https://github.com/kubevirt/monitoring/tree/main/test/metrics/prom-metrics-linter, see the readme as well it's basically the [promlint](https://github.com/prometheus/client_golang/blob/v1.16.0/prometheus/testutil/promlint/promlint.go), and we added that the metric should start with kubevirt_hco/cdi/ssp/hpp/..)

Counter do need to end with total, maybe the start should be `kubernetes_nmstate`?

Reply:
I agree with 'nmstate_topology_applied_total'